cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst

```diff
@@ -256,87 +272,256 @@ The response will be the same JSON array with additional parameters for each of
+    {"appId": "App2", "programType":"Flow", "programId":"Flow1", "statusCode":404, "error": "App: App2 not found"}
+  ]
+
+In this particular example, the service and mapreduce programs in the *App1* application were successfully
+started, and there was an error starting the last program because the *App2* application does not exist. 
+
+.. _http-restful-api-lifecycle-stop:
 
-.. rubric::  Examples
+Stop a Program
+--------------
+You can stop the flows, MapReduce programs, schedules, Spark programs, workflows,
+workers, and custom services of an application by submitting an HTTP POST request::
+
+  POST <base-url>/namespaces/<namespace>/apps/<app-id>/<program-type>/<program-id>/stop
 
 .. list-table::
    :widths: 20 80
-   :stub-columns: 1
+   :header-rows: 1
 
-   * - HTTP Method
-     - ``POST <base-url>/namespaces/default/apps/HelloWorld/flows/WhoFlow/start``
-   * - Description
-     - Start a flow *WhoFlow* in the application *HelloWorld* in the namespace *default*
-   * - 
-     - 
-   * - HTTP Method
-     - ``POST <base-url>/namespaces/default/apps/Count/services/GetCounts/stop``
-   * - Description
-     - Stop the service *GetCounts* in the application *Count* in the namespace *default*
-   * - 
-     - 
-   * - HTTP Method
-     - ``GET <base-url>/namespaces/default/apps/HelloWorld/flows/WhoFlow/status``
-   * - Description
-     - Get the status of the flow *WhoFlow* in the application *HelloWorld* in the namespace *default*
-   * - 
-     - 
-   * - HTTP Method
-     - ``POST <base-url>/namespaces/default/status``
-   * - HTTP Body
-     - ``[{"appId": "MyApp", "programType": "flow", "programId": "MyFlow"},``
-       ``{"appId": "MyApp2", "programType": "service", "programId": "MyService"}]``
-   * - HTTP Response
-     - ``[{"appId":"MyApp", "programType":"flow", "programId":"MyFlow", "status":"RUNNING", "statusCode":200},``
-       ``{"appId":"MyApp2", "programType":"service", "programId":"MyService",``
-       ``"error":"Program not found", "statusCode":404}]``
-   * - Description
-     - Attempt to get the status of the flow *MyFlow* in the application *MyApp* and of the service *MyService*
-       in the application *MyApp2* in the namespace *default*
+   * - Parameter
+     - Description
+   * - ``<namespace>``
+     - Namespace ID
+   * - ``<app-id>``
+     - Name of the application being called
+   * - ``<program-type>``
+     - One of ``flows``, ``mapreduce``, ``schedules``, ``spark``, ``workflows``, ``workers``, or ``services``
+   * - ``<program-id>``
+     - Name of the *flow*, *MapReduce*, *schedule*, *Spark*, *workflow*, *worker*, or *custom service*
+       being called
 
-When starting an program, you can optionally specify runtime arguments as a JSON map in the request body::
+A program that is in the STOPPED state cannot be stopped. If there are multiple runs of the program
+in the RUNNING state, this call will stop one of the runs, but not all of the runs. 
```

Comment:
I'm not sure, but dont think we should document it even if we know.  This behavior is also weird, seems like we should stop them all.